keyboards/centromere/keymaps/mini/keymap.c

```diff
@@ -0,0 +1,161 @@
+  KC_NO,   KC_CAPS, KC_TILD, _______, _______, _______,      _______, _______, _______, KC_PIPE,  KC_DQT, KC_NO,   \
+                    _______, _______, _______,      KC_ENT,  _______, KC_DEL                    \
+),
+
+/* Adjust (Lower + Raise)
+ *
+ * ,----------------------------------.           ,----------------------------------.
+ * |  F1  |  F2  |  F3  |  F4  |  F5  |           |   F6 |  F7  |  F8  |  F9  |  F10 |
+ * |------+------+------+------+------|           |------+------+------+------+------|
+ * |  F11 |  F12 |      |      |      |           |      |      |      |Taskmg|caltde|
+ * |------+------+------+------+------|           |------+------+------+------+------|
+ * | Reset|      |      |      |      |           |      |      |      |      |      |
+ * `----------------------------------'           `----------------------------------'
+ *                  ,--------------------.    ,------,-------------.
+ *                  |      | LOWER|      |    |      | RAISE|      |
+ *                  `-------------|      |    |      |------+------.
+ *                                |      |    |      |
+ *                                `------'    `------'
+ */
+[_ADJUST] =  LAYOUT( \
+  KC_NO,   KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,        KC_F6,   KC_F7,   KC_F8,   KC_F9,   KC_F10,   KC_NO,   \
+  KC_NO,   KC_F11,  KC_F12,  _______, _______, _______,      _______, _______, _______, TSKMGR, CALTDEL,   KC_NO,   \
+  KC_NO,   RESET,   _______, _______, _______, _______,      _______, _______, _______, _______,  _______, KC_NO,   \
+                    _______, _______, _______,      _______,  _______, _______                    \
+)
+};
+
+void persistant_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        #ifdef AUDIO_ENABLE
+          PLAY_SONG(tone_qwerty);
+        #endif
+        persistant_default_layer_set(1UL<<_QWERTY);
+      }
+      return false;
+      break;
+    case LOWER:
+      if (record->event.pressed) {
+        layer_on(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_LOWER);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+      break;
+    case RAISE:
+      if (record->event.pressed) {
+        layer_on(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      } else {
+        layer_off(_RAISE);
+        update_tri_layer(_LOWER, _RAISE, _ADJUST);
+      }
+      return false;
+      break;
+    case ADJUST:
+      if (record->event.pressed) {
+        layer_on(_ADJUST);
+      } else {
+        layer_off(_ADJUST);
+      }
+      return false;
+      break;
+  }
+  return true;
+}
```

Comment:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) {
  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
}
```

You don't really need the persistent layer set function because `QWERTY` custom keycode is never used in the keymap, and you only have one usable "default" layer.

The `process_record_user` function is only handling the adjust layer, which can be replaced entirely with the succinct snippet in this suggestion and a few additional `define`s up top.

Reply:
You don’t need to commit this suggestion, but as-is you have defined for the MO keycodes (which will work with the suggestion here) and you have cases in process_record_user (which would need the enum’d custom keycodes you had before that the MO defines replaced).

I don’t have strong feelings with either method, I just figured you could achieve things more simply and in fewer lines with this suggestion.  But you will have to pick one or the other method.

Let me know if you need clarification.

Reply:
@ridingqwerty  The simpler the better. Which defines are needed? I can't commit this suggestion because it says it's outdated.. I've been spoiled by the "commit suggestion" feature.

Reply:
You already have the defines need to use the `layer_state_set_user` method.

I believe my original suggestion is outdated because Drashna's suggestion to remove the QWERTY case from process_record_user was a smaller subset of mine.  I'll re-suggest to see if you'd be able to just use the "commit suggestion".